### ISSUE TITLE
Weird issues with child classes

### ISSUE THREAD

**Issue by ukch** [Benjaminsen/extend.js#9]
So, I have a class that inherits from another class. Both of these classes have a 'child' class that is referenced in the outer 'extends' class (sorry, don't know what it's called!).
Anyway, somewhere in the constructors, these classes get combined into one ugly lump. Check out the simplified example below: (also on [jsfiddle](http://jsfiddle.net/phzwud26/))

```javascript
// <ul></ul> in HTML
var ul = $("ul");

var Child = Class.extend(function() {
    var self = this;
    self.doSomething = function() {
        $("<li>").text("Child doing something").appendTo(ul);
    };
});

var Parent1 = Class.extend(function() {
    var self = this;
    self.ChildClass = Child;
    self.constructor = function() {
        $("<li>").text("Parent1 constructor").appendTo(ul);;
        self.child = new self.ChildClass();
    };
    self.doSomething = function() {
        $("<li>").text("Parent1 doing something").appendTo(ul);;
        self.child.doSomething();
    }
});

var Parent2 = Parent1.extend(function() {
    var self = this;
    self.constructor = function() {
        self.super();
        $("<li>").text("Parent2 constructor").appendTo(ul);;
    }
});

new Parent2().doSomething();
/*
Expected result:
* Parent1 constructor
* Parent2 constructor
* Parent1 doing something
* Child doing something

Actual result:
* Parent1 constructor
* Parent2 constructor
* Child doing something
*/
```

**Comment by Benjaminsen** [comment]
Quite interesting, due to the way the scope system works in ExtendJS the scope where destroyed when created from a reference stored in another class.

This bug is fixed in 0.2.3 
[jsfiddle fork](http://jsfiddle.net/hvmxutz6/) using 0.2.3.

**Comment by ukch** [comment]
The previous link is still using 0.2.2. [This link](http://jsfiddle.net/hvmxutz6/1/) is using the new version.
